Update how-to-auto-train-image-models.md (#111816)
Azure/azure-content: articles/machine-learning/how-to-auto-train-image-models.md

```
---

## Choose orchestration (preview)
> [!IMPORTANT]
> This feature is currently in public preview. This preview version is provided without a service-level agreement. Certain features might not be supported or might have constrained capabilities. For more information, see [Supplemental Terms of Use for Microsoft Azure Previews](https://azure.microsoft.com/support/legal/preview-supplemental-terms/).

# [Azure CLI](#tab/cli)

[!INCLUDE [cli v2](../../includes/machine-learning-cli-v2.md)]

You can choose to use the legacy runtime or the new components (in preview) which run on [Azure Machine Learning pipelines](concept-ml-pipelines.md). You can indicate that you want to use the pipeline orchestration by setting `_automl_subgraph_orchestration` flag to `"true"` in the job properties. For example,

```yaml
properties:
  _automl_subgraph_orchestration: "true"
```

# [Python SDK](#tab/python)

[!INCLUDE [sdk v2](../../includes/machine-learning-sdk-v2.md)]

> You can choose to use the legacy runtime or the new components (in preview) which run on [Azure Machine Learning pipelines](concept-ml-pipelines.md). You can indicate that you want to use the pipeline orchestration by setting `_automl_subgraph_orchestration` flag to `"true"` in the job properties. For example,

Reviewer: legacy runtime or the new components --> This might not be very clear to the users, what do we mean by legacy runtime.

Reviewer: @bhimar - what's our thought on it?

#### Supported model architectures

The following table summarizes the supported legacy models for each computer vision task. Using only these legacy models will trigger runs using the legacy runtime (where each individual run or trial is submitted as a command job).

> 

Reviewer 2: I'd add "Please see below for HuggingFace and MMDetection support"

Object detection | **YOLOv5**: One stage object detection model   <br>  **Faster RCNN ResNet FPN**: Two stage object detection models  <br> **RetinaNet ResNet FPN**: address class imbalance with Focal Loss <br> <br>*Note: Refer to [`model_size` hyperparameter](reference-automl-images-hyperparameters.md#model-specific-hyperparameters) for YOLOv5 model sizes.*| ***`yolov5`\**** <br> `fasterrcnn_resnet18_fpn` <br> `fasterrcnn_resnet34_fpn` <br> `fasterrcnn_resnet50_fpn` <br> `fasterrcnn_resnet101_fpn` <br> `fasterrcnn_resnet152_fpn` <br> `retinanet_resnet50_fpn` 
Instance segmentation | **MaskRCNN ResNet FPN**| `maskrcnn_resnet18_fpn` <br> `maskrcnn_resnet34_fpn` <br> ***`maskrcnn_resnet50_fpn`\****  <br> `maskrcnn_resnet101_fpn` <br> `maskrcnn_resnet152_fpn`

> 

Reviewer 2: I would add another section marked preview:

#### Supported model architectures - HuggingFace, MMDetection

> With the new AutoML on Pipelines feature (preview), you can additionally use any image classification model from the [HuggingFace Hub](https://huggingface.co/models?pipeline_tag=image-classification&library=transformers) which are part of the transformers library (such as microsoft/beit-base-patch16-224), as well as any object detection or instance segmentation model from the [MMDetection Version 2.28.2 Model Zoo](https://mmdetection.readthedocs.io/en/v2.28.2/model_zoo.html) (such as atss_r50_fpn_1x_coco). Using any HuggingFace or MMDetection model will trigger runs using pipeline components. If both legacy and HuggingFace/MMdetection models are used, all runs/trials will be triggered using components. Curated models from HuggingFace and MMDetection are provided in the azureml-staging registry. These curated models have been tested thoroughly and use default hyperparameters selected from extensive benchmarking to ensure effective training. The table below summarizes these curated models

Reviewer 2: "With the new backend that runs on Pipelines"

> With the new AutoML on Pipelines feature (preview), you can additionally use any image classification model from the [HuggingFace Hub](https://huggingface.co/models?pipeline_tag=image-classification&library=transformers) which are part of the transformers library (such as microsoft/beit-base-patch16-224), as well as any object detection or instance segmentation model from the [MMDetection Version 2.28.2 Model Zoo](https://mmdetection.readthedocs.io/en/v2.28.2/model_zoo.html) (such as atss_r50_fpn_1x_coco). Using any HuggingFace or MMDetection model will trigger runs using pipeline components. If both legacy and HuggingFace/MMdetection models are used, all runs/trials will be triggered using components. Curated models from HuggingFace and MMDetection are provided in the azureml-staging registry. These curated models have been tested thoroughly and use default hyperparameters selected from extensive benchmarking to ensure effective training. The table below summarizes these curated models

Reviewer 2: I would start a new paragraph with:

In addition to supporting any model from HuggingFace and MMDetection, we also offer a list of curated models from these libraries in azureml-staging registry. These curated models have been tested thoroughly and use default hyperparameters selected from extensive benchmarking to ensure effective training. The table below summarizes these curated models

Task |  model architectures
---|---------

> Image classification<br> (multi-class and multi-label)| [**microsoft/beit-base-patch16-224-pt22k-ft22k**](https://ml.azure.com/registries/azureml-preview/models/microsoft-beit-base-patch16-224-pt22k-ft22k/version/1): The BEiT model is a Vision Transformer (ViT), which is a transformer encoder model (BERT-like). <br> [**google/vit-base-patch16-224**](https://ml.azure.com/registries/azureml-preview/models/google-vit-base-patch16-224/version/1): The Vision Transformer (ViT) is a transformer encoder model (BERT-like) pretrained on a large collection of images in a supervised fashion, namely ImageNet-21k, at a resolution of 224x224 pixels. <br> [**facebook/deit-base-patch16-224**](https://ml.azure.com/registries/azureml-preview/models/facebook-deit-base-patch16-224/version/1): This model is actually a more efficiently trained Vision Transformer (ViT). <br> [**microsoft/swinv2-base-patch4-window12-192-22k**](https://ml.azure.com/registries/azureml-preview/models/microsoft-swinv2-base-patch4-window12-192-22k/version/1): The Swin Transformer is a type of Vision Transformer.

Reviewer 2: Let's have the same table format as above - with the additional colum for the string syntax, and only BEIT name in the first

Object Detection | TODO
Instance Segmentation | TODO

> You can also get a list of the curated models for a given task using the Python SDK:

Reviewer 2: We constantly update the list of curated models. You can get the most up to date list of the curated models for a given task using the Python SDK:

```
credential = DefaultAzureCredential()
ml_client = MLClient(credential, registry_name="azureml-staging")

models = ml_client.models.list()
classification_models = []
for model in models:
    model = ml_client.models.get(model.name, label="latest")
    if model.tags['task'] == 'image-classification': # choose an image task
        classification_models.append(model.name)

classification_models
```
output:
```
['google-vit-base-patch16-224',
 'microsoft-swinv2-base-patch4-window12-192-22k',
 'facebook-deit-base-patch16-224',
 'microsoft-beit-base-patch16-224-pt22k-ft22k']

> ```

Reviewer 2: I would move here the phrase: "Using any HuggingFace or MMDetection model will trigger runs using pipeline components. If both legacy and HuggingFace/MMdetection models are used, all runs/trials will be triggered using components."

The users should not care much about this part, at least not until they already read the above

Object detection | **YOLOv5**: One stage object detection model   <br>  **Faster RCNN ResNet FPN**: Two stage object detection models  <br> **RetinaNet ResNet FPN**: address class imbalance with Focal Loss <br> <br>*Note: Refer to [`model_size` hyperparameter](reference-automl-images-hyperparameters.md#model-specific-hyperparameters) for YOLOv5 model sizes.*| ***`yolov5`\**** <br> `fasterrcnn_resnet18_fpn` <br> `fasterrcnn_resnet34_fpn` <br> `fasterrcnn_resnet50_fpn` <br> `fasterrcnn_resnet101_fpn` <br> `fasterrcnn_resnet152_fpn` <br> `retinanet_resnet50_fpn` 
Instance segmentation | **MaskRCNN ResNet FPN**| `maskrcnn_resnet18_fpn` <br> `maskrcnn_resnet34_fpn` <br> ***`maskrcnn_resnet50_fpn`\****  <br> `maskrcnn_resnet101_fpn` <br> `maskrcnn_resnet152_fpn`

#### Supported model architectures - HuggingFace and MMDetection (preview)

> With the new backend that runs on [AzureML Pipelines](concept-ml-pipelines.md), you can additionally use any image classification model from the [HuggingFace Hub](https://huggingface.co/models?pipeline_tag=image-classification&library=transformers) which is part of the transformers library (such as microsoft/beit-base-patch16-224), as well as any object detection or instance segmentation model from the [MMDetection Version 2.28.2 Model Zoo](https://mmdetection.readthedocs.io/en/v2.28.2/model_zoo.html) (such as atss_r50_fpn_1x_coco). 

Reviewer 3: ```suggestion
With the new backend that runs on [Azure Machine Learning pipelines](concept-ml-pipelines.md), you can additionally use any image classification model from the [HuggingFace Hub](https://huggingface.co/models?pipeline_tag=image-classification&library=transformers) which is part of the transformers library (such as microsoft/beit-base-patch16-224), as well as any object detection or instance segmentation model from the [MMDetection Version 2.28.2 Model Zoo](https://mmdetection.readthedocs.io/en/v2.28.2/model_zoo.html) (such as atss_r50_fpn_1x_coco). 
```
Fixing terminology

In addition to supporting any model from HuggingFace Transfomers and MMDetection 2.28.2, we also offer a list of curated models from these libraries in the azureml-staging registry. These curated models have been tested thoroughly and use default hyperparameters selected from extensive benchmarking to ensure effective training. The table below summarizes these curated models.

Task |  model architectures | String literal syntax
---|----------|----------

> Image classification<br> (multi-class and multi-label)| **BEiT**: A Vision Transformer (ViT), which is a transformer encoder model (BERT-like). |[`microsoft/beit-base-patch16-224-pt22k-ft22k`](https://ml.azure.com/registries/azureml-preview/models/microsoft-beit-base-patch16-224-pt22k-ft22k/version/1)<br> [`google/vit-base-patch16-224`](https://ml.azure.com/registries/azureml-preview/models/google-vit-base-patch16-224/version/1)<br> [`facebook/deit-base-patch16-224`](https://ml.azure.com/registries/azureml-preview/models/facebook-deit-base-patch16-224/version/1)<br> [`microsoft/swinv2-base-patch4-window12-192-22k`](https://ml.azure.com/registries/azureml-preview/models/microsoft-swinv2-base-patch4-window12-192-22k/version/1)

Reviewer 2: ![image](https://github.com/MicrosoftDocs/azure-docs/assets/3809388/75d62815-3508-4035-a3ae-acd226a56671)

missing the description for other models apart from Beit - we should have one line for each model family like for the legacy ones; also, not sure we should put the links, these are not very useful for automl users

Object Detection | TODO |  `sparse_rcnn_r50_fpn_300_proposals_crop_mstrain_480-800_3x_coco `<br> `sparse_rcnn_r101_fpn_300_proposals_crop_mstrain_480-800_3x_coco` <br> `deformable_detr_twostage_refine_r50_16x2_50e_coco` <br> `vfnet_r50_fpn_mdconv_c3-c5_mstrain_2x_coco` <br> `vfnet_x101_64x4d_fpn_mdconv_c3-c5_mstrain_2x_coco` <br> `yolof_r50_c5_8x8_1x_coco` <br> `mask_rcnn_swin-t-p4-w7_fpn_fp16_ms-crop-3x_coco`
Instance Segmentation | TODO | TODO

We constantly update the list of curated models. You can get the most up to date list of the curated models for a given task using the Python SDK:
```
credential = DefaultAzureCredential()

> ml_client = MLClient(credential, registry_name="azureml-staging")

Reviewer: why staging?

Author: azureml-staging is the registry where the models will be for public preview

> We constantly update the list of curated models. You can get the most up to date list of the curated models for a given task using the Python SDK:

Reviewer 3: ```suggestion
We constantly update the list of curated models. You can get the most up-to-date list of the curated models for a given task using the Python SDK:
```
Fixing punctuation

models = ml_client.models.list()
classification_models = []
for model in models:
    model = ml_client.models.get(model.name, label="latest")
    if model.tags['task'] == 'image-classification': # choose an image task
        classification_models.append(model.name)

classification_models
```

> output:

Reviewer 3: ```suggestion
Output:
```
Fixing capitalization

Task |  model architectures | String literal syntax
---|----------|----------

> Image classification<br> (multi-class and multi-label)| **BEiT**: A Vision Transformer (ViT), which is a transformer encoder model (BERT-like). <br> **ViT**: A BERT-like transformer encoder model which is pretrained on a large collection of images in a supervised fashion. <br> **DeiT**: A more efficiently trained Vision Transformer (ViT) <br> **SwinV2**: A type of Vision Transformer used in both image classification and dense recognition tasks | `microsoft/beit-base-patch16-224-pt22k-ft22k`<br> `google/vit-base-patch16-224`<br> `facebook/deit-base-patch16-224`<br> `microsoft/swinv2-base-patch4-window12-192-22k`

Reviewer 2: I am wondering if instead of providing the descriptions that we do here, which I honestly don't find very useful, we can just make the names of the models links to HF model cards, like https://huggingface.co/microsoft/swinv2-base-patch4-window12-192-22k

Author: I think that is a good idea! I don't think the descriptions are very useful either, there isn't much detail we can put in a table.

Reviewer: Should we instead point them to azureml registry and/or model catalogue?

Author: I think it makes sense to point users to the registry model cards for curated models, which also have links to the HF model cards. We would need to point directly to azureml-staging registry though, as model catalogue pulls from azureml (prod) registry.

#### Supported model architectures

The following table summarizes the supported legacy models for each computer vision task. These models are compatible with the legacy runtime orchestration as well as the new pipelines orchestration.

> If you are using the pipelines orchestration, you can additionally use image classification models from the [HuggingFace Hub](https://huggingface.co/models?pipeline_tag=image-classification&library=transformers) which are part of the transformers library (such as microsoft/beit-base-patch16-224), as well as object detection and instance segmentation models from the [MMDetection Model Zoo](https://mmdetection.readthedocs.io/en/latest/model_zoo.html) (such as atss_r50_fpn_1x_coco). For curated models from HuggingFace and MMDetection, please visit the [Azure Machine Learning Model Catalog](concept-foundation-models.md).

Reviewer 2: we should list curated models explictly and say what happens for those - like default hps, we made sure they work etc

Author: Please let me know what you think of the current table. If it looks good I'll start populating the OD and IS entries. We will need to update when the models are pushed to azureml staging

> If you are using the pipelines orchestration, you can additionally use image classification models from the [HuggingFace Hub](https://huggingface.co/models?pipeline_tag=image-classification&library=transformers) which are part of the transformers library (such as microsoft/beit-base-patch16-224), as well as object detection and instance segmentation models from the [MMDetection Model Zoo](https://mmdetection.readthedocs.io/en/latest/model_zoo.html) (such as atss_r50_fpn_1x_coco). For curated models from HuggingFace and MMDetection, please visit the [Azure Machine Learning Model Catalog](concept-foundation-models.md).

Reviewer 2: use image classification models => use any image classification models

same for mmdetection; also specify the version of mmdetection we support

> If you are using the pipelines orchestration, you can additionally use image classification models from the [HuggingFace Hub](https://huggingface.co/models?pipeline_tag=image-classification&library=transformers) which are part of the transformers library (such as microsoft/beit-base-patch16-224), as well as object detection and instance segmentation models from the [MMDetection Model Zoo](https://mmdetection.readthedocs.io/en/latest/model_zoo.html) (such as atss_r50_fpn_1x_coco). For curated models from HuggingFace and MMDetection, please visit the [Azure Machine Learning Model Catalog](concept-foundation-models.md).

Reviewer: Instead of https://mmdetection.readthedocs.io/en/latest/model_zoo.html, point tot 2.28.2 version's link
